packages/alchemy/src/react/hooks/usePlugin.ts

```diff
@@ -0,0 +1,139 @@
+import { useMutation } from "@tanstack/react-query";
+import type { Chain, Client, Transport } from "viem";
+import type { SupportedAccount } from "../../config";
+import type { UseSmartAccountClientResult } from "./useSmartAccountClient";
+
+export type UsePluginProps<
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined,
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  }
+> = {
+  client?: UseSmartAccountClientResult<
+    TTransport,
+    TChain,
+    SupportedAccount<"MultiOwnerModularAccount">
```

Comment:
if we make this generalizable then this should be the union type of supported accounts